actions/image_partition_action.go

```diff
@@ -413,14 +415,24 @@ func (i ImagePartitionAction) Run(context *debos.DebosContext) error {
 
 	context.ImageMntDir = path.Join(context.Scratchdir, "mnt")
 	os.MkdirAll(context.ImageMntDir, 0755)
-	for _, m := range i.Mountpoints {
+	for idx, m := range i.Mountpoints {
 		dev := i.getPartitionDevice(m.part.number, *context)
 		mntpath := path.Join(context.ImageMntDir, m.Mountpoint)
+
+		// check if the recipe already mounted something at mountpoint
+		for _, mntpoint := range i.Mountpoints {
+			if (mntpoint.Mountpoint == m.Mountpoint && mntpoint.mounted) {
+				return fmt.Errorf("%s mount failed: mountpoint at %s already exists", m.part.Name, m.Mountpoint)
```

Comment:
This should be done in Verify instead; Such we can fail early before spending minutes in the recipe :) Also just error out if there are duplicated mountpoints whether it was mounted is irrelevant